src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+  const props = sourceString.split(';').map((prop) => prop.trim());
+
+  for (const prop of props) {
+    const [key, value] = prop.split(':');
+
+    if (key) {
+      obj[key.trim()] = value.trim();
```

Comment:
There is a potential issue if the `value` is undefined or missing. Consider adding a check to handle such cases, possibly assigning a default value or skipping the property.